server/publish-counts.js

```diff
@@ -76,13 +76,13 @@ Counts.publish = function(self, name, cursor, options) {
   }
 
   if (!countFn) {
-    self.added('counts', name, {count: cursor.count()});
+    self.added('counts', name, {count: await cursor.countAsync()});
```

Comment:
I think this will have to use `Collection.countDocuments` instead in Meteor 3. Either way, better migrate to the new API here if possible.

Reply:
I'm afraid that this is not possible without affecting the logic.

Cursors do not support `countDocuments`, only collections. The entirety of `Counts.publish = async function(self, name, cursor, options) { ... }` revolves around manipulating the cursor of the publication in whose context `Counts` does its own publishing. We would have to introduce logic that gets the collection from the cursor and then does `Collection.countDocuments([selector])`. We would even have to hack our way around extracting the selector from somewhere.

I suggest we leave it like that and update to `Cursor.count()` when Meteor to gets rid of `*Async` APIs.

Reply:
I found a way, by reusing some of the internal APIs of the `mongo` package. Thankfully, the `Cursor` object [stores the collection name, selector, and options in `_cursorDescriptor`](https://github.com/meteor/meteor/blob/release-3.0/packages/mongo/mongo_driver.js#L840) , which are then trivial to reuse in the `countDocuments` method of the `_mongo` object. 

This package has in fact always been relying on `_cursorDescriptor`. If this property gets removed in the future by Meteor, then `publish-counts` may have to go altogether the way of the Dodo. But for now I think it will survive.

Note that for local collections we fetch all the documents and return the length of the array.